### DISCUSSION TITLE
Unexpectedly high memory usage

### DISCUSSION THREAD

**Discussion by hrsr0** [voxeltycoon/issues#572] [General]
**Describe the bug**
memory leak, massive memory leak

**Steps to reproduce**
see attached image

issue present in 0.71/0.72/0.73

**Build version**
0.71/0.72/0.73 are affected 

**Platform**
win 10 pro x64

**Reproducibility**
see attached image
[memory leak after 3-4 hours of playing ](https://user-images.githubusercontent.com/52865925/61175475-c9d11880-a5b8-11e9-8fae-7b9c4af13dbb.jpg)

**Comment by hrsr0** [comment]
tested @ 1270x720 ALL the time in window mode, and with default settings to eliminate any discrepancies since day 1,  severe gamebreaking issue.

**Comment by katepolak** [comment]
First off, this game is written in C# so there can't be memory leaks.

Second of all, as far as I know nobody has encountered an issue like this before and I know for a fact that people have more advanced maps than that, so it seems to be an issue with your machine. I myself have never had memory issues and I only have 8GB of RAM, it was fine even after tens of hours on a single save.

As for the memory increase, those are most likely just simply due to you building stuff not the autosave. This could be easily tested by just letting it run while doing nothing though. That's honestly the only test I can think of. Logs would be appreciated too if possible.

**Comment by hurzelchen** [comment]
Having a garbage collector doesn't mean you can't have memory leaks.

And having a garbage collector doesn't mean it has to run. Maybe hrsr0's machine has enough RAM that it saw no need to free memory?

**Comment by katepolak** [comment]
What I mean is that there is no way for developers to do anything about this as it's all handled by the garbage collector.

He has 16GB of RAM so not that different from most people who play the game, there must be something else causing the issue. As I said before, there isn't anything I can do with the information we have right now and the devs can't do anything either because as far as I know it wasn't successfully reproduced yet.

**Comment by Toys0125** [comment]
well, I went and did some testing on my own with a map with nothing but generated towns that total around 5700Mb when loaded and used free cam up high to make sure all the terrain was loaded.
I would do this back and forward and messaged my memory each run after giving 30 secs in the main menu for the garbage collector to run.
1925.5Mb
2057.8Mb
2088.8Mb
2178.2Mb
2214.9Mb
So each run would slowly increase after each load. But this is unity we are talking about so it could be an engine thing.
